src/Formatter/FormatterTrait.php

```diff
@@ -212,4 +195,49 @@ protected function getParsedPattern(NumberFormat $numberFormat, $style)
      * @return string[] The patterns, keyed by style.
      */
     abstract protected function getAvailablePatterns(NumberFormat $numberFormat);
+
+    /**
+     * Returns the replacements to be used for the localizeNumber method.
+     *
+     * Can be overriden which is useful for deviating symbols for currencies.
+     *
+     * @param NumberFormat $numberFormat
+     *
+     * @return array
+     */
+    protected function getLocalizeReplacements(NumberFormat $numberFormat): array
```

Comment:
It feels odd to define these two methods in FormatterTrait only to have them immediately overridden in CurrencyFormatter.
I suggest making them abstract here, and then implemented in both CurrencyFormatter and NumberFormatter. That also saves you from having to explain the overriding ("Can be overriden which is useful...")

Reply:
I would also rename getLocalizeReplacements() to either getLocalizedReplacements() or even getLocalizedSymbols(). The second option would require renaming getParseReplacements() as well to something like getCanonicalSymbols(), dunno if that's an improvement.

Reply:
Agreed about the "feels odd". I was considering your proposed solution making it abstract in the trait and wasn't sure whether having a trait with an abstract method feels even more odd, but overall, I prefer having the abstract trait method.

Regarding naming, I think your two suggestions are improvements and I've applied them.